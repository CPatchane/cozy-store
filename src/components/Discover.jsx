import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
