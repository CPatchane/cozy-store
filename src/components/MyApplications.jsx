import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
