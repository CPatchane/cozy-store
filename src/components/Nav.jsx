import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
