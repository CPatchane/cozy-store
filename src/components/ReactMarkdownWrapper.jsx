import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
