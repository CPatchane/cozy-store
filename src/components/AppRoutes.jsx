import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
