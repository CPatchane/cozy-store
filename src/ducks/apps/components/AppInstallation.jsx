/* global cozy */
import React, { Component } from 'react'

import { ModalContent, ModalHeader, ModalFooter } from 'cozy-ui/react/Modal'
import Spinner from 'cozy-ui/react/Spinner'

import PermissionsList from './PermissionsList'
import ReactMarkdownWrapper from '../../components/ReactMarkdownWrapper'

import { getLocalizedAppProperty } from 'ducks/apps'
import { translate } from 'cozy-ui/react/I18n'

class AppInstallation extends Component {
  installApp() {
    this.setState({ error: null })
    const { app, onError, onSuccess, channel } = this.props
    this.props
      .installApp(app.slug, app.type, channel)
      .then(() => {
        onSuccess(app)
      })
      .catch(error => {
        if (onError) return onError(error)
        throw error
      })
  }

  render() {
    const {
      app,
      fetchError,
      installError,
      isFetching,
      isInstalling,
      isCanceling,
      lang,
      onCancel,
      t
    } = this.props

    const appName = getLocalizedAppProperty(app, 'name', lang)
    const permissions = app.permissions || {}
    const isFirstLoading = isFetching && !isCanceling

    return (
      <div className="sto-install">
        <ModalHeader className="sto-install-header">
          <h2>{t('app_modal.install.title')}</h2>
        </ModalHeader>
        {isFirstLoading
          ? <ModalContent>
            <div className="sto-install-loading">
              <Spinner size="xlarge" />
            </div>
          </ModalContent>
          : <ModalContent>
            {permissions && (
              <div>
                <ReactMarkdownWrapper
                  source={t('app_modal.install.permissions.description', {
                    cozyName: cozy.client._url.replace(/^\/\//, ''),
                    appName: app.name
                  })}
                />
                <PermissionsList
                  app={app}
                  permissions={app.permissions}
                  appName={appName}
                />
              </div>
            )}
            {fetchError && (
              <p className="u-error">
                {t('app_modal.install.message.version_error', {
                  message: fetchError.message
                })}
              </p>
            )}
          </ModalContent>
        }
        {!isFirstLoading && !fetchError && (
          <ModalFooter>
            {installError && (
              <p className="u-error">
                {t('app_modal.install.message.install_error', {
                  message: installError.message
                })}
              </p>
            )}
            <div className="sto-install-controls">
              <button
                role="button"
                className="c-btn c-btn--secondary"
                onClick={onCancel}
                disabled={isInstalling || isCanceling}
                aria-busy={isCanceling}
              >
                <span>{t('app_modal.install.cancel')}</span>
              </button>
              <button
                role="button"
                disabled={isInstalling || isCanceling}
                aria-busy={isInstalling}
                className="c-btn c-btn--regular c-btn--download"
                onClick={() => this.installApp()}
              >
                <span>{t('app_modal.install.install')}</span>
              </button>
            </div>
          </ModalFooter>
        )}
      </div>
    )
  }
}

export default translate()(AppInstallation)
