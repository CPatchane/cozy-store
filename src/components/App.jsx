
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
