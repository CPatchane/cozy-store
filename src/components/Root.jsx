import React from 'react'
import { Provider } from 'react-redux'
import { I18n } from 'cozy-ui/react/I18n'
import { HashRouter } from 'react-router-dom'

import App from './App'

const Root = ({ context, lang, store }) => {
  return <I18n lang={lang} dictRequire={(lang) => require(`../locales/${lang}`)}>
    <Provider store={store}>
      <HashRouter>
        <App />
      </HashRouter>
    </Provider>
  </I18n>
}

export default Root
