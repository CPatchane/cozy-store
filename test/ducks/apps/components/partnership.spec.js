'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from '../../../jestLib/I18n'
import { Partnership } from 'ducks/apps/components/Partnership'

import mockAppVersion from '../_mockPhotosRegistryVersion'

const appWithPartnership = Object.assign({}, mockAppVersion, {
  partnership: {
    icon: '/registry/mockapp/1.0.0/partnership_icon',
    description: 'A legal for this mock partnership'
  }
})

describe('Partnership component', () => {
  it('should be rendered correctly the parternship informations', () => {
    const component = shallow(
      <Partnership t={tMock} app={appWithPartnership} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
