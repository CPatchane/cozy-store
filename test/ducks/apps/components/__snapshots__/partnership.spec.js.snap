// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partnership component should be rendered correctly the parternship informations 1`] = `
<div
  className="sto-partnership"
>
  <img
    className="sto-partnership-icon"
    onError={[Function]}
    src="/registry/mockapp/1.0.0/partnership_icon"
    style={Object {}}
  />
  <span>
    <ReactMarkdownWrapper
      source="A legal for this mock partnership"
    />
  </span>
</div>
`;
